app/models/user.rb

```diff
@@ -12,6 +12,12 @@ class User < ApplicationRecord
     denied: "denied"
   }, _prefix: true
 
+  enum access_level: {
```

Comment:
Why role this up into user?  Here are the benefits to keeping it on the resource:

* More clear separation of concerns -- we don't add yet more data or behavior to our User model
* Makes it easy to allow different access levels per resource, which has been discussed often.
* Makes it easier to name the related "access related thing" -- when it has the role on it, `user.access` makes sense and is a coherent thing in our domain. When we split it off, we now have a really awkward name of `accessible_resource` and yet `access_level` lives on the User. The fact that both talk about `access_` is a tip off that this access modeling is cohesive.

The only advantage to adding it to the user is that it enforces the 'one access level per user', but you already have that via the uniqueness validation on the access model.

Reply:
I largely agree with your point, I think there are trade-offs, but I think the reasons for not doing this are slightly different for me, I've tried to address them [here](https://github.com/simpledotorg/simple-server/pull/1185/files#r466342832).